test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -142,18 +142,18 @@ namespace integration_framework {
                          ->on_proposal();
 
     proposals
-        .filter([](auto proposal) {
-          return boost::size(proposal->transactions()) != 0;
+        .filter([](auto event) {
+          return boost::size(event.proposal->get()->transactions()) != 0;
         })
-        .subscribe([this](auto proposal) {
-          proposal_queue_.push(proposal);
+        .subscribe([this](auto event) {
+          proposal_queue_.push(*event.proposal);
           log_->info("proposal");
           queue_cond.notify_all();
         });
 
     auto proposal_flat_map =
         [](auto t) -> rxcpp::observable<std::tuple_element_t<0, decltype(t)>> {
-      if (boost::size(std::get<1>(t)->transactions()) != 0) {
+      if (boost::size(std::get<1>(t).proposal->get()->transactions()) != 0) {
```

Comment:
`get->get.get->*()`

I wonder is it possible to get simpler :D

by the way, it is ok now